test/Microsoft.Azure.Devices.Edge.Test.Common/linux/EdgeDaemon.cs

```diff
@@ -17,37 +17,48 @@ public class EdgeDaemon : IEdgeDaemon
 
         public static async Task<EdgeDaemon> CreateAsync(CancellationToken token)
         {
-            string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", token);
-            if (platformInfo.Length == 1)
+            SupportedPackageExtension packageExtension;
+            string os, version;
+            string[] platformInfo = await Process.RunAsync("cat", @"/etc/os-release", token);
+            os = Array.Find(platformInfo, element => element.StartsWith("ID="));
+            version = Array.Find(platformInfo, element => element.StartsWith("VERSION_ID="));
+
+            // VERSION_ID is desired but it is an optional field
+            if (version == null)
             {
-                platformInfo = platformInfo[0].Split(' ');
+                version = Array.Find(platformInfo, element => element.StartsWith("VERSION="));
             }
 
-            string os = platformInfo[0].Trim();
-            string version = platformInfo[1].Trim();
-            SupportedPackageExtension packageExtension;
+            if (os == null || version == null)
+            {
+                throw new NotImplementedException($"Failed to gather operating system information (OS: '{os}', Version: '{version}')");
```

Comment:
You're trying to print `{os}` and `{version}` when one or both of them are null. Won't that cause a null-ref exception?

Reply:
Thanks Damon!

I'm not a c# programmer, I thought it would print out as null. I've fixed that and ensured that potential double quotes are trimmed from the version and os variables (which can potentially happen according to what Bear and me have seen, e.g. the CentOS we're using).

Reply:
Actually, it might just print an empty string...I don't remember.